src/time.rs

```diff
@@ -0,0 +1,170 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Instant {
+    millis: u64,
+}
+
+impl Instant {
+    /// Create a new `Instant`.
+    pub fn new(millis: u64) -> Instant {
+        Instant { millis }
+    }
+
+    /// Add a [Duration] to the current timestamp value and return a
+    /// `Option` containing a new [Instant] of the result on success.
+    ///
+    /// [Duration]: struct.Duration.html
+    /// [Instant]: struct.Instant.html
+    pub fn checked_add(&self, rhs: Duration) -> Option<Instant> {
+        self.millis.checked_add(rhs.millis()).map(|x| Instant::new(x))
+    }
```

Comment:
Like I mentioned, there is no condition in which `checked_add` will ever return `None`. This function is meaningless.